src/Enzo/enzo_EnzoRiemannUtils.hpp

```diff
@@ -0,0 +1,263 @@
+// See LICENSE_CELLO file for license and copyright information
+
+/// @file     enzo_EnzoRiemannUtils.hpp
+/// @author   Matthew Abruzzo (matthewabruzzo@gmail.com)
+/// @date     Thurs May 15 2020
+/// @brief    [\ref Enzo] Declaration and Implementation of assorted utilities
+/// in the enzo_riemann_utils namesapce that are associated with the
+/// implementation of Riemann Solvers.
+///
+/// Many of these functions could conceivably be methods of EnzoRiemannImpl,
+/// but doing so would cause the compiler to generate unnecessary duplicate
+/// versions of of template functions for the different ImplFunctors that use
+/// the same underlying LUT. It might be worth considering making these
+/// functions into static methods of EnzoRiemannLUT.
+
+#ifndef ENZO_ENZO_RIEMANN_UTILS_HPP
+#define ENZO_ENZO_RIEMANN_UTILS_HPP
+
+//----------------------------------------------------------------------
+
+namespace enzo_riemann_utils{
+
```

Comment:
Should these functions also be inlined?

Reply:
I'm happy to mark them all as inline.

As an aside, a little while back, I had done an experiment where I made extremely liberal use of the `always_inline` function attribute with gcc and it slowed some things down considerably. (Although that's not particularly relevant to the current situation; the compiler can always elect to ignore the `inline` specification if it thinks inlining will slow things down)

Reply:
I'm not surprised, especially for `gcc`. We've seen some cases with the Intel compiler where the compiler wasn't inlining when it would actually be faster. Intel takes ages to compile though with all the optimizations.